ccm_web/server/src/canvas/canvas.interfaces.ts

```diff
@@ -1,17 +1,35 @@
+import { CanvasOAuthReturnQueryDto } from './dtos/canvas.oauth.query.dto'
 import { hasKeys } from '../typeUtils'
 
 // OAuth
 
 export interface OAuthGoodResponseQuery {
   code: string
   state: string
+  error: undefined
+  error_description: undefined
 }
+
 export interface OAuthErrorResponseQuery {
   error: string
+  code: undefined
```

Comment:
why the type is undefined here?

Reply:
I guess I was just trying to make explicit that we expect `code` to not be defined when there is an error.

Reply:
I don't think so in this case? Since if it's an error, we expect it to be `undefined`. Making it `any` would negate type checking on that property, and `unknown` I think would be similar to not including it in the interface at all.